template.yaml

```diff
@@ -28,6 +28,12 @@ Parameters:
     Default: ""
     AllowedPattern: '|subnet-[a-z0-9]+'
 
+  NeptuneDBInstanceClass:
+    Description: The DBInstanceClass to be used for the Neptune Database
+    Type: String
+    Default: db.serverless
+    AllowedPattern: ^db\.(serverless|[a-z0-9]+\.[a-z0-9]+)$
```

Comment:
for the db instance class param, a user could enter an incorrect class name and the template wil build and fail right? would it be a better experience to add allowedvalues instead?

Reply:
Correct a user could input something that would cause the stack creation to fail. Yes we could solve this by using allowed values. However the list is big and complicated. So i deliberately decided not to do this as not only would it be a big list but we would have to keep it up to date and even then it is also sometimes dependant on the engine version. So my view was to balance the chance of failure with keeping it simple so went with this.